app/assets/v2/js/grants/ingest-missing-contributions.js

```diff
@@ -98,52 +106,45 @@ Vue.component('grants-ingest-contributions', {
     },
 
     async ingest(event) {
+      let txHash;
+      let userAddress;
+
       try {
         event.preventDefault();
-
-        // Return if form is not valid
         const formParams = this.checkForm();
 
         if (!formParams) {
-          return;
+          return; // return if form is not valid
         }
 
         // Make sure wallet is connected
-        let walletAddress;
+        const walletAddress = web3 ? (await web3.eth.getAccounts())[0] : undefined;
```

Comment:
Should we be triggering  web3modal to be sure people is using it and connected to the site instead?

Reply:
If they don't have a wallet, this will throw an error and they can just connect by clicking in the header. But if you prefer to manually trigger web3modal that's fine too. Let me know.

If you do want web3modal, is this the correct way to do it?

```javascript
initWallet();
await onConnect();
const walletAddress = (await web3.eth.getAccounts())[0];
```

Reply:
yes that should work, is just a detail to be sure the UX helps the users to understand they are connected with the right wallet and help them understand if not working. 
to get the web3modal address 
`const walletAddress = (await selectedAccount); `